spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseDeleteOrphanFilesSparkAction.java

```diff
@@ -361,4 +403,90 @@ static PathFilter forSpecs(Map<Integer, PartitionSpec> specs) {
       return partitionNames.isEmpty() ? HiddenPathFilter.get() : new PartitionAwareHiddenPathFilter(partitionNames);
     }
   }
+
+  private static Dataset<Row> sanitizeAndSplitPaths(Dataset<Row> filesDF) {
+    return filesDF.mapPartitions((MapPartitionsFunction<Row, Row>) input ->
```

Comment:
Since Spark is already going to convert from `UnsafeRow` to whatever this requires, I would much prefer to use a Java class instead of `Row`. That way we get type checking in Spark and don't need to call `row.getString(0)` anywhere.

Reply:
I actually think we should offer methods that return `Dataset<String>` for valid paths in the base class. That way, we can remove one round of unnecessary conversion to and from internal Spark representation. Right now, we have a dataset of string, which we convert to a dataset of rows and immediately convert back to access strings. Spark won't optimize this, unfortunately.

The idea of using a bean to carry the metadata sounds promising. I'd give it a try.